ext/spl/spl_array.c

```diff
@@ -1878,6 +1888,26 @@ SPL_METHOD(Array, __unserialize)
 	}
 
 	object_properties_load(&intern->std, Z_ARRVAL_P(members_zv));
+
+	if (iterator_class_zv && Z_TYPE_P(iterator_class_zv) != IS_NULL) {
+		if (Z_TYPE_P(iterator_class_zv) != IS_STRING) {
+			zend_throw_exception_ex(spl_ce_UnexpectedValueException, 0,
+				"Expected string or null at index 3 of serialized Array data; got %s",
+				zend_get_type_by_const(Z_TYPE_P(iterator_class_zv)));
+			return;
+		} else {
+			zend_class_entry *ce = zend_lookup_class(Z_STR_P(iterator_class_zv));
+
+			if (ce) {
```

Comment:
We'll also have to check that `ce` is instanceof `Iterator` here.

Reply:
OK, will do.

Reply:
Thanks for being so thorough. It's appreciated.

Reply:
Done.